runtime/util/hshelp.c

```diff
@@ -2643,8 +2643,8 @@ recreateRAMClasses(J9VMThread * currentThread, J9HashTable * classHashTable, J9H
 		if (!fastHCR) {
 			vmFuncs->hashClassTableDelete(classLoader, J9UTF8_DATA(className), J9UTF8_LENGTH(className));
 			if ((NULL != vm->sharedClassConfig) && (NULL != vm->sharedClassConfig->romToRamHashTable)) {
-				omrthread_rwmutex_enter_write(vm->sharedClassConfig->romToRamHashTableMutex);
 				RomToRamEntry entry;
+				omrthread_rwmutex_enter_write(vm->sharedClassConfig->romToRamHashTableMutex);
 				entry.ramClass = originalRAMClass;
```

Comment:
Suggest swapping these two lines (initializing `entry` before acquiring the mutex).